python/lsst/ts/GenericCamera/driver/canoncamera.py

```diff
@@ -169,17 +177,89 @@ async def endReadout(self):
-        tags = {
-            "TOP": 0,
-            "LEFT": 0,
-            "WIDTH": self.width,
-            "HEIGHT": self.height,
-            "EXPOSURE": self.exposure_time,
-            "ISO": self.iso,
-        }
+        await self._set_tag_values()
+
         image = exposure.Exposure(
-            luminance, self.width, self.height, tags, isJPEG=False
+            luminance, self.width, self.height, self.tags, isJPEG=False
         )
         return image
+
+    async def _set_tag_values(self):
+        """Convenience coroutine to provide values for all the tags in the FITS
+        header."""
+        # ---- Date, night and basic image information ----
+        now_string = datetime.datetime.now(tz=datetime.timezone.utc).strftime(
+            genericcamera.DATETIME_FORMAT
+        )
+        date_obs = self.datetime_start.strftime(genericcamera.DATE_FORMAT)
+        date_beg = self.datetime_start.strftime(genericcamera.DATETIME_FORMAT)
+        date_end = self.datetime_end.strftime(genericcamera.DATETIME_FORMAT)
+        self.get_tag(name="DATE").value = now_string
+        self.get_tag(name="DATE-OBS").value = date_obs
+        self.get_tag(name="DATE-BEG").value = date_beg
+        self.get_tag(name="DATE-END").value = date_end
+        self.get_tag(name="MJD").value = Time(now_string, format="mjd").value
+        self.get_tag(name="MJD-OBS").value = Time(date_obs, format="mjd").value
+        self.get_tag(name="MJD-BEG").value = Time(date_beg, format="mjd").value
+        self.get_tag(name="MJD-END").value = Time(date_end, format="mjd").value
+        # TODO Not sure what value to set here.
+        self.get_tag(name="OBSID").value = ""
+        self.get_tag(name="IMGTYPE").value = "OBJECT"
+
+        # ---- Pointing info, etc. ----
+        # Always pointing at the zenith.
+        elevation = 90.0
+        # Minor axis always points south
+        azimuth = 0.0
+
+        # Retrieve observing location info
+        lon = next((tag for tag in self.tags if tag.name == "OBS-LONG")).value
+        lat = next((tag for tag in self.tags if tag.name == "OBS-LAT")).value
+        height = next((tag for tag in self.tags if tag.name == "OBS-ELEV")).value
+        # Create EarthLocation instance for Rubin Observatory
+        rubin = EarthLocation.from_geodetic(lon=lon, lat=lat, height=height)
+
+        radec_start = self.__get_radec_from_altaz_location_time(
+            alt=elevation, az=azimuth, obs_time=date_beg, location=rubin
+        )
+        radec_end = self.__get_radec_from_altaz_location_time(
+            alt=elevation, az=azimuth, obs_time=date_end, location=rubin
+        )
+        self.get_tag(name="RASTART").value = radec_start.ra.value
+        self.get_tag(name="DECSTART").value = radec_start.dec.value
+        self.get_tag(name="RAEND").value = radec_end.ra.value
+        self.get_tag(name="DECEND").value = radec_end.dec.value
+        # Rotation is always zero.
+        self.get_tag(name="ROTPA").value = 0
```

Comment:
Why is that ROTPA always zero? is that because you are assuming this is only used for the all sky?

Reply:
Correct. I expended the comment to reflect that.